airbyte-integrations/connectors-performance/destination-harness/src/main/java/io/airbyte/integrations/destination_performance/PerformanceTest.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination_performance;
+
+import static java.lang.Thread.sleep;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.airbyte.commons.features.EnvVariableFeatureFlags;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.config.AllowedHosts;
+import io.airbyte.config.EnvConfigs;
+import io.airbyte.config.ResourceRequirements;
+import io.airbyte.config.WorkerDestinationConfig;
+import io.airbyte.protocol.models.AirbyteMessage;
+import io.airbyte.protocol.models.AirbyteMessage.Type;
+import io.airbyte.protocol.models.AirbyteRecordMessage;
+import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
+import io.airbyte.workers.WorkerConfigs;
+import io.airbyte.workers.internal.DefaultAirbyteDestination;
+import io.airbyte.workers.internal.exception.DestinationException;
+import io.airbyte.workers.process.AirbyteIntegrationLauncher;
+import io.airbyte.workers.process.KubePortManagerSingleton;
+import io.airbyte.workers.process.KubeProcessFactory;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import java.io.BufferedReader;
+import java.io.FileOutputStream;
+import java.io.FileReader;
+import java.net.InetAddress;
+import java.net.URL;
+import java.nio.channels.Channels;
+import java.nio.channels.FileChannel;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.regex.Pattern;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class PerformanceTest {
+
+  public static final int PORT1 = 9877;
+  public static final int PORT2 = 9878;
+  public static final int PORT3 = 9879;
+  public static final int PORT4 = 9880;
+
+  public static final Set<Integer> PORTS = Set.of(PORT1, PORT2, PORT3, PORT4);
+
+  public static final double MEGABYTE = Math.pow(1024, 2);
+  private final String imageName;
+  private final JsonNode config;
+  private final ConfiguredAirbyteCatalog catalog;
+  private final String dataSourceURL;
+
+  private DefaultAirbyteDestination destination;
+
+  PerformanceTest(final String imageName, final String config, final String catalog, final String datasource) throws JsonProcessingException {
+    final ObjectMapper mapper = new ObjectMapper();
+    this.imageName = imageName;
+    this.config = mapper.readTree(config);
+    this.catalog = Jsons.deserialize(catalog, ConfiguredAirbyteCatalog.class);
+    this.dataSourceURL = datasource;
+  }
+
+  void runTest() throws Exception {
+    KubePortManagerSingleton.init(PORTS);
```

Comment:
nit: runTest() is too long, probably benefits from breaking up into smaller private functions.

Reply:
at minimum, I think we can break out the logging code and the reader loop into separate functions.

Reply:
We can also break out the code to read the data source into memory.